### PR TITLE
Add initial model commits

### LINKED ISSUE #11: New Models and Model Testing
- Add models based on UML Diagram
- Conduct testings for Models

CHIM Class Diagram

### PR DESCRIPTION
**Pull request by daniellau88**
Add new models needed for CHIM

### CODE REVIEW

File: src/main/java/seedu/address/model/Date.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+        .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+        .toFormatter(Locale.ENGLISH);
+
+    public static final DateTimeFormatter TO_STRING_FORMATTER = DateTimeFormatter.ofPattern("yyyy-M-d HH:mm");
+
+    public final LocalDateTime value;
+
+    /**
+     * Constructs a {@code Phone}.
```
Comment (weixue123) [inline src/main/java/seedu/address/model/Date.java]: Need to update the documentation here

File: src/main/java/seedu/address/model/cheese/Cheese.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package seedu.address.model.cheese;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+/**
+ * Represents a Cheese in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
```
Comment (weixue123) [inline src/main/java/seedu/address/model/cheese/Cheese.java]: Update documentation

File: src/main/java/seedu/address/model/cheese/ExpiryDate.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import java.time.LocalDateTime;
+
+import seedu.address.model.Date;
+
+/**
+ * Represents a Cheese's expiry date in the address book.
+ * Guarantees: immutable
+ */
+public class ExpiryDate extends Date {
+    /**
```
Comment (weixue123) [inline src/main/java/seedu/address/model/cheese/ExpiryDate.java:15]: Add a vertical spacing here? - the auto-formatter should take care of this
Same issue with ManufactureDate, CompletedDate and OrderDate

File: src/main/java/seedu/address/model/order/Order.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import seedu.address.model.cheese.Cheese;
+import seedu.address.model.cheese.CheeseType;
+
+public class Order {
+    private static int currentId = 0;
+
+    // Identity fields
+    // Primary key for Order
+    private final int orderId;
+    private final CheeseType cheeseType;
```
Comment (weixue123) [inline src/main/java/seedu/address/model/order/Order.java]: If Order has a cheeseType property, then I would assume that one order should only have one type of cheese. Then, I think there should be some checks to ensure this?

An alternative would be to let an Order have multiple cheese types. If so, then we will have to remove the cheeseType property.

File: src/main/java/seedu/address/model/order/Order.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    public CheeseType getCheeseType() {
+        return cheeseType;
+    }
+
+    public int getOrderId() {
+        return orderId;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
```
Comment (weixue123) [inline src/main/java/seedu/address/model/order/Order.java]: Update documentation

File: src/main/java/seedu/address/model/order/Order.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.address.model.cheese.Cheese;
+import seedu.address.model.cheese.CheeseType;
+
```
Comment (weixue123) [inline src/main/java/seedu/address/model/order/Order.java:12]: Add documentation

File: src/main/java/seedu/address/model/order/Quantity.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
+package seedu.address.model.order;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
```
Comment (weixue123) [inline src/main/java/seedu/address/model/order/Quantity.java:5]: Is this class used anywhere yet?

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 8]
```diff
@@ -60,6 +60,19 @@
     public static final EditCommand.EditPersonDescriptor DESC_AMY;
     public static final EditCommand.EditPersonDescriptor DESC_BOB;
```
Comment (weixue123) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:62]: I believe these should be placed under testutil instead (not logic/commands/CommandTestUtil.

Also, it will be great if we tested invalid dates and cheese types as well.

File: src/test/java/seedu/address/model/cheese/CheeseTest.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+ */
+public class CheeseTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new CheeseBuilder().withCheeseType(null).build());
+    }
+
+    @Test
+    public void isSameCheese() {
+        // same object -> return true
```
Comment (weixue123) [inline src/test/java/seedu/address/model/cheese/CheeseTest.java]: Standardize all comments to start with capital letter?

**Review by weixue123 — COMMENTED** [review]
Minor comments on documentation, and one major one regarding the design of the Order class.

Other than that, looks good!

File: src/main/java/seedu/address/model/order/Order.java (hunk 4 above)
Comment (daniellau88) [reply src/main/java/seedu/address/model/order/Order.java]: Will bring up a proposal for this in the meeting

File: src/main/java/seedu/address/model/order/Quantity.java (hunk 7 above)
Comment (daniellau88) [reply src/main/java/seedu/address/model/order/Quantity.java:5]: My bad, the class should be used in `Order`, will add in in next commit

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 8 above)
Comment (daniellau88) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:62]: 3 Things
1. This portion of code is to test the `EditPersonDescriptor` (which is not a part of the model, but only used inside `Commands`, hence this should be in the right place
1. Regarding invalid dates, will add a few in the Dates class
1. Regarding invalid cheese types, currently there is no way for a cheese to be of an invalid type (because it just needs a name), unless we check for empty String

File: src/test/java/seedu/address/model/cheese/CheeseTest.java (hunk 9 above)
Comment (daniellau88) [reply src/test/java/seedu/address/model/cheese/CheeseTest.java]: Would be nice to standardize but might not be necessary (will have a lot of existing codes to update)

Will bring it up in meeting tmr

File: src/main/java/seedu/address/model/order/Quantity.java (hunk 7 above)
Comment (AhQuanz) [reply src/main/java/seedu/address/model/order/Quantity.java:5]: This is used in line 20 to ensure validity of quantity

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 8 above)
Comment (AhQuanz) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:62]: Will be adding test cases for empty String

**Review by weixue123 — APPROVED** [review]
